packages/autorest.typescript/test/unit/transforms/parameterTransforms.spec.ts

```diff
@@ -422,7 +422,7 @@ describe("parameterTransforms", () => {
       assert.equal(p1.parameter, param1);
     });
 
-    it("should not guard non-required query parameters name", () => {
+    it("should guard non-required query parameters name", () => {
```

Comment:
Is the mismatch between mapper and property name from required parameter?

Do we have a case for that?